docs/contributing/design-documents.rst

```diff
@@ -0,0 +1,74 @@
+:title: Design Documents
+:description: Information necessary for a feature design document.
+
+.. _design-documents:
+
+Design Documents
+================
+
+Before submitting a pull request which is expected to significantly change the
+behavior of Deis (such as when contributing a new feature or refactor),
+contributors should first open an issue representing a design document.
+
+Goals
+-----
+
+Design documents were adopted in order to:
+
+* Involve stakeholders as early as possible in a feature's development
+* Ensure code changes accomplish the original motivations and design goals
+* Establish clear acceptance criteria for a feature or change
+* Enforce test-driven design methodology and automated test coverage
+
+Contents
+--------
+
+Design document issues should be named ``Design Doc: <change description>`` and
```

Comment:
I think it's fine to keep them as separate issues for now, though we can experiment with in-line design docs in the future if we're failing to get feedback. :)